### PR TITLE
feat(*): remove capability signing

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This series of commits removes capability signing from various crates.

Since wasmCloud is now ideally WIT-driven, we do not need to specify or sign our providers or actors with capabilities and can deduce them from the related WIT contracts as necessary.

### CODE REVIEW

File: crates/host/src/wasmbus/mod.rs [hunk 1]
```diff
@@ -1317,7 +1317,7 @@ impl ComponentSpecification {
 
 #[allow(clippy::large_enum_variant)] // Without this clippy complains actor is at least 0 bytes while provider is at least 280 bytes. That doesn't make sense
 enum Claims {
-    Actor(jwt::Claims<jwt::Actor>),
+    Actor(jwt::Claims<jwt::Component>),
```
Comment (thomastaylor312) [inline crates/host/src/wasmbus/mod.rs:1320]: Should we rename this variant to Component while we're here?

File: crates/runtime/src/actor/mod.rs [hunk 2]
```diff
@@ -6,23 +6,27 @@ pub use component::{
...
     pub require_signature: bool,
 }
 
-/// Extracts and validates claims contained within `WebAssembly` binary, if such are found
-fn claims(wasm: impl AsRef<[u8]>) -> anyhow::Result<Option<jwt::Claims<jwt::Actor>>> {
+/// Extracts and validates claims contained within a WebAssembly binary, if present
+///
+/// # Arguments
+///
+/// * `wasm` - Bytes that consitute a valid WebAssembly binary
```
Comment (thomastaylor312) [inline crates/runtime/src/actor/mod.rs:24]: Spelling: constitute

**Review by thomastaylor312 — COMMENTED** [review]
Overall the code looks great. However, I do have a question around providers. Providers might have wit that they use to generate their code, but they don't carry wit with them like a component. So how is a provider supposed to indicate what it is allowed to do?

File: crates/host/src/wasmbus/mod.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/host/src/wasmbus/mod.rs:1320]: Ah yes, will rename!

- [ ] rename `Claims::Actor` variant

File: crates/runtime/src/actor/mod.rs (hunk 2 above)
Comment (vados-cosmonic) [reply crates/runtime/src/actor/mod.rs:24]: - [ ] fix spelling in rustdoc

**Comment by vados-cosmonic** [comment]
> However, I do have a question around providers. Providers might have wit that they use to generate their code, but they don't carry wit with them like a component.

So it's possible for a provider to carry around the WIT that it was generated with, but IIRC this was argued against for the initial implementation of wRPC. There's no reason we can't have this information if we want. 

> So how is a provider supposed to indicate what it is allowed to do?

Right now, the invocation lookup would be the only way -- it's parametrized with a lattice name, target ID and wRPC version, but you can find out exactly what world key names and functions are used.

That said, we're actually not restricting this in any way right now, so a plan on what/how to restrict probably makes sense before we add that in.

**Review by thomastaylor312 — APPROVED** [review]

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**